python/generate.py

```diff
@@ -95,6 +95,19 @@ def get_function_docstring_dict(func, funcname, signature_prefix):
                 return_obj['description'] = parse_rst(returns.description) if returns.description else ''
                 return_obj['return_name'] = returns.return_name
                 description['returns'].append(return_obj)
+        
+        description['source'] = []
+        repo_prefix = "https://github.com/streamlit/streamlit/blob/develop/lib"
+
+        try: # For Streamlit commands wrapped by decorator
+            path_parts = func.__wrapped__.__code__.co_filename.partition("/streamlit")
+            line_number = func.__wrapped__.__code__.co_firstlineno
```

Comment:
@snehankekre : May want to do a `while` loop checking whether there's a `.__wrapped__` . In fact, you could even remove the try/except this way:

```
while '__wrapped__' in func:
  func = func.__wrapped__

path_parts = func.__code__.co_filename.partition("/streamlit")
line_number = func.__code__.co_firstlineno
```

Reply:
So much cleaner, thank you! I did:

```
while '__wrapped__' in func.__dict__:
  func = func.__wrapped__

path_parts = func.__code__.co_filename.partition("/streamlit")
line = func.__code__.co_firstlineno
```